src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+package com.epam.page.object.generator.integrationalTests;
+
+import com.epam.page.object.generator.PageObjectsGenerator;
+import com.epam.page.object.generator.adapter.JavaPoetAdapter;
+import com.epam.page.object.generator.containers.SupportedTypesContainer;
+import com.epam.page.object.generator.parser.JsonRuleMapper;
+import com.epam.page.object.generator.utils.XpathToCssTransformation;
+import com.epam.page.object.generator.validators.ValidatorsStarter;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.IOFileFilter;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import javax.tools.JavaCompiler;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.ToolProvider;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class MainIntegrationTest {
+    private static final String RESOURCE_DIR = "src/test/resources/";
+    private static final String TEST_DIR = "test/";
+    private static final String MANUAL_DIR = "manual/";
+    private static final String TEST_CLASS_PACKAGE_NAME = "test.";
+    private static final String MANUAL_CLASS_PACKAGE_NAME = "manual.";
+    private static final String PACKAGE_TEST_NAME = "test";
+    private static final IOFileFilter IO_FILE_FILTER = new IOFileFilter() {
+        @Override
+        public boolean accept(File file) {
+            return false;
+        }
+
+        @Override
+        public boolean accept(File dir, String name) {
+            return false;
+        }
+    };
+
+    private boolean checkLocatorUniqueness;
+    private boolean forceGenerateFiles;
+    private String inputJavaFileTest;
+    private String inputJavaFileManual;
+    private String generatedClassFullName;
+    private String manualClassFullName;
+    private String jsonSourcePath;
+    private String url;
+    private Class testClass;
+    private Class manualClass;
+
+    private static List<File> filesList = new ArrayList<>();
```

Comment:
this variable can be moved into method data()

Reply:
fixed.